packages/core/src/controllers/subscriber.ts

```diff
@@ -109,6 +111,33 @@ export class Subscriber extends ISubscriber {
     }
   };
 
+  public isSubscribed: ISubscriber["isSubscribed"] = async (topic: string) => {
+    // topic subscription is already resolved
+    if (this.topics.includes(topic)) return true;
+
+    // wait for the subscription to resolve
+    return await new Promise((resolve, reject) => {
+      const watch = new Watch();
+      watch.start(this.pendingSubscriptionWatchLabel);
+
+      const interval = setInterval(() => {
+        if (!this.pending.has(topic) && this.topics.includes(topic)) {
+          clearInterval(interval);
+          watch.stop(this.pendingSubscriptionWatchLabel);
+          resolve(true);
+        }
+        if (
+          watch.elapsed(this.pendingSubscriptionWatchLabel) / 1000 >=
```

Comment:
Would it make sense to define the `TIMEOUT` constant in ms (`FIVE_SECONDS * 1000`) to avoid the inline division here?